internal/pkg/addon/storage.go

```diff
@@ -147,12 +159,13 @@ func NewDDBTemplate(input *DynamoDBProps) *DynamoDBTemplate {
 
 // RDSProps holds RDS-specific properties for addon.NewRDSTemplate().
 type RDSProps struct {
-	WorkloadType   string   // The type of the workload associated with the RDS addon.
-	ClusterName    string   // The name of the cluster.
-	Engine         string   // The engine type of the RDS Aurora Serverless cluster.
-	InitialDBName  string   // The name of the initial database created inside the cluster.
-	ParameterGroup string   // The parameter group to use for the cluster.
-	Envs           []string // The copilot environments found inside the current app.
+	WorkloadType      string   // The type of the workload associated with the RDS addon.
+	ClusterName       string   // The name of the cluster.
+	ServerlessVersion string   // The version of Aurora Serverless
```

Comment:
I wonder if this is a hint that we need two types of constructors? and can replace the [existing `RDSTemplate` constructor](https://github.com/hkford/copilot-cli/blob/8a2ab8eb528dac097901ffc8ae2967772f535610/internal/pkg/addon/storage.go#L172):

```go
type RDSProps struct {
   WorkloadType
   // Other existing fields...
   Envs []string
   
   serverlessVersion string
}

func NewServerlessV1Template(rds RDSProps) *RDSTemplate {
  rds.serverlessVersion = auroraServerlessVersionV1
  // existing logic...
}

func NewServerlessV2Template(rds RDSProps) *RDSTemplate {
   rds.serverlessVersion = auroraServerlessVersionV2
}
```

Reply:
I agree with you. I made two types of constructors.